### PR TITLE
feat(@desktop/wallet): Provide QProperties for token management

### PR DESCRIPTION
**Pull request by Cuteivist**
task #12725

corresponding status-go change: https://github.com/status-im/status-go/pull/4391

### What does the PR do

* Provided QProperty group by community checkbox
* Provided QProperty to fetch token preferences
* Provided slot to update token preferences

Token preferences are communicated via Json, because they will be passed to new C++ Controller.

### Affected areas

Settings / Token management

### Screenshot of functionality (including design for comparison)

No UI was added

### CODE REVIEW

File: src/app/modules/main/wallet_section/all_tokens/controller.nim [hunk 1]
```diff
@@ -74,3 +80,18 @@ proc getTokensDetailsLoading*(self: Controller): bool =
 
 proc getTokensMarketValuesLoading*(self: Controller): bool =
   self.tokenService.getTokensMarketValuesLoading()
+
+proc updateTokenPreferences*(self: Controller, tokenPreferencesJson: string) =
+  self.tokenService.updateTokenPreferences(tokenPreferencesJson)
+
+proc getTokenPreferencesJson*(self: Controller): string =
+  let data = self.tokenService.getTokenPreferences()
+  if data == nil:
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/all_tokens/controller.nim:89]:
```suggestion
  if data.isNil:
```

File: src/app_service/service/token/service.nim [hunk 2]
```diff
@@ -604,3 +609,31 @@ QtObject:
...
+    try:
+      let preferencesJson = parseJson(tokenPreferencesJson)
+      var tokenPreferences: seq[TokenPreferences]
+      if preferencesJson.kind == JArray:
+        for preferences in preferencesJson:
+          add(tokenPreferences, fromJson(preferences, TokenPreferences))
+      let response = backend.updateTokenPreferences(tokenPreferences)
+      if not response.error.isNil:
+        error "status-go error", procName="updateTokenPreferences", errCode=response.error.code, errDesription=response.error.message
+        return
```
Comment (saledjenic) [inline src/app_service/service/token/service.nim:634]:
```suggestion
        raise newException(CatchableError, response.error.message)
```

File: src/app/modules/main/wallet_section/all_tokens/view.nim [hunk 3]
```diff
@@ -128,3 +128,28 @@ QtObject:
   proc currencyFormatsUpdated*(self: View) =
     self.flatTokensModel.currencyFormatsUpdated()
     self.tokensBySymbolModel.currencyFormatsUpdated()
+
+  proc tokenPreferencesUpdated*(self: View, result: bool) {.signal.}
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/all_tokens/view.nim:132]: @Cuteivist I don't see any interaction with the qml for this and all below added signals/slots?

File: src/app/modules/main/wallet_section/all_tokens/controller.nim (hunk 1 above)
Comment (saledjenic) [reply src/app/modules/main/wallet_section/all_tokens/controller.nim]: The previous implementation was not wrong, I just made a suggestion. ;) Thanks.

**Review by saledjenic — APPROVED** [review]

File: src/app_service/service/token/service.nim [hunk 4]
```diff
@@ -604,3 +609,30 @@ QtObject:
...
+
+  proc updateTokenPreferences*(self: Service, tokenPreferencesJson: string) =
+    var updated = false
+    try:
+      let preferencesJson = parseJson(tokenPreferencesJson)
+      var tokenPreferences: seq[TokenPreferences]
+      if preferencesJson.kind == JArray:
+        for preferences in preferencesJson:
+          add(tokenPreferences, fromJson(preferences, TokenPreferences))
+      let response = backend.updateTokenPreferences(tokenPreferences)
```
Comment (Khushboo-dev-cpp) [inline src/app_service/service/token/service.nim:631]: @Cuteivist  I have a question why are we saving this under the settings db? or creating a new settings for this? would that be more inline with how we manage settings variables

Comment (Khushboo-dev-cpp) [reply src/app_service/service/token/service.nim:631]: got it. I understand now so its saved along with the tokens. as its a setting per token.

**Review by Khushboo-dev-cpp — APPROVED** [review]
LGTM :)

**Pull request merged**